tests/stavenote_tests.ts

```diff
@@ -467,16 +468,16 @@ function drawBoundingBoxes(options: TestOptions, contextBuilder: ContextBuilder)
   ];
   options.assert.expect(noteStructs.length * 2);
 
+  const notes: StaveNote[] = [];
   for (let i = 0; i < noteStructs.length; ++i) {
-    const note = draw(
-      staveNote(noteStructs[i]),
-      stave,
-      ctx,
-      (i + 1) * 25,
-      true /* drawBoundingBox */,
-      false /* addModifierContext */
-    );
+    notes.push(staveNote(noteStructs[i]));
+  }
+  notes[2].addModifier(new Ornament('tr'), 0);
+  notes[2].addModifier(new Accidental('b'), 0);
+  Dot.buildAndAttach([notes[2]], { all: true });
```

Comment:
I always thought this API didn't fit the rest of the VexFlow API. I recall you and I settled on this due to problems with circular references.... is that correct? I wonder if we can improve the Dot.buildAndAttach() API in V5, and maybe make it more similar to the other ways of adding modifiers, e.g.:

```
notes[2].addModifier(new Dot())
```

Can we investigate this in a separate PR? I am curious if we no longer have the circular reference problem.

Reply:
For example, the v3.0.9 API looked like:

```
note.addDot(index)
note.addDotToAll()
```

https://github.com/0xfe/vexflow/blob/00ec15c67ff333ea49f4d3defbd9e22374c03684/src/stavenote.js#L810-L823

It would be nice if we could restore that API, or at least make it look like `note.addModifier(new Dot())`

Reply:
note.addModifier(new Dot()) would not work for chords. You need one Dot per NoteHead.

Reply:
OK, we can discuss this in a separate issue / PR. I think the older 3.0.9 API of addDot(index) or addDotToAll() was very friendly and easy to understand. If we can restore that API without running into any circular reference problems, I think we should.